services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/multipart/SdkPojoConversionUtils.java

```diff
@@ -225,4 +307,47 @@ private static Map<String, Object> retrieveSdkFields(SdkPojo sourceObject, List<
                                     field.getValueOrDefault(sourceObject)),
             Map::putAll);
     }
+
+    private static void setSdkFields(SdkPojo targetBuilder, SdkPojo sourceObject) {
+        setSdkFields(targetBuilder, sourceObject, new HashSet<>());
+    }
+
+    private static void setSdkFields(SdkPojo targetBuilder, SdkPojo sourceObject, Set<String> fieldsToIgnore) {
+        Map<String, Object> sourceFields = retrieveSdkFields(sourceObject, sourceObject.sdkFields());
+        List<SdkField<?>> targetSdkFields = targetBuilder.sdkFields();
+
+        for (SdkField<?> field : targetSdkFields) {
+            if (fieldsToIgnore.contains(field.memberName())) {
+                continue;
+            }
+            field.set(targetBuilder, sourceFields.getOrDefault(field.memberName(), null));
+        }
+    }
+
+    private static void validateRequestFields(SdkPojo sourceObject, SdkPojo targetBuilder, Set<String> allowedFields) {
+        Set<String> invalidFields = new HashSet<>();
+
+        for (SdkField<?> sourceField : sourceObject.sdkFields()) {
+            String fieldName = sourceField.memberName();
+            Object sourceValue = sourceField.getValueOrDefault(sourceObject);
+
+            if (!allowedFields.contains(fieldName)) {
+                SdkField<?> targetField = targetBuilder.sdkFields()
+                                                       .stream()
+                                                       .filter(field -> field.memberName().equals(fieldName))
+                                                       .findFirst()
+                                                       .orElse(null);
+                if (targetField != null && !targetField.getValueOrDefault(targetBuilder).equals(sourceValue)) {
+                    invalidFields.add(fieldName);
+                }
+            }
+        }
+
+        if (!invalidFields.isEmpty()) {
+            throw SdkClientException.create(
+                String.format("The following fields are not allowed: %s",
```

Comment:
Are there currently any fields not on the allow list and do we know if any of them have different default values?  IE - is there a risk of now raising an exception for cases that was previously working for customers?